script/install.sh

```diff
@@ -1292,11 +1297,28 @@ case $(uname -m) in
         fatal "Unsupported Arch $(uname -m)"
         ;;
 esac
+if [[ ${ARCH_OVERRIDE:+1} ]]; then
+    ARCH=$ARCH_OVERRIDE
+fi
 if [[ $OS-$ARCH = windows-x86_64 && $COMPILER = ldc* ]]; then
     ARCH=x64
 fi
-if [[ $OS-$ARCH = osx-aarch64 && $COMPILER = ldc* ]]; then
-    ARCH=arm64
+if [[ $OS-$ARCH = osx-aarch64 ]]; then
+    if [[ $COMPILER == dmd* ]]; then
+        log "
+DMD does not have builds for macOS on aarch64/arm64 achitecture (e.g. for the M1 CPU).
```

Comment:
I would recommend against mentioning `M1`. There are already other CPUs than M1.

Reply:
removed it

Reply:
There are others but there are no others worth buying, so I think we should try to be somewhat welcoming to M1 users in this regard - probably not here but we could use a "how to use D on apple silicon" page on dlang.org since it apparently isn't obvious.

Reply:
> There are others but there are no others worth buying,

What's wrong with M1 Pro and M1 Max? Too expensive?

> so I think we should try to be somewhat welcoming to M1 users in this regard

In general I'm against selecting out specific CPU models. I don't see any mentioning of specific Intel, AMD or Via CPU models. I've seen other open source projects that provide binaries for "Intel" or for "M1". But M1 is not an architecture, it's a CPU model. If the user has M1 Max, does that count as a M1 binary? What about the next CPU model, that would most likely be called M2? The official name of the architecture that Apple uses when targeting end users is "Apple Silicon". When targeting developers, they use "ARM64". Anything else is just going to cause confusion. I don't want the D community to do the same misstake. So far, LDC has been using "ARM64", which is perfectly fine since it's targeting developers.

> probably not here but we could use a "how to use D on apple silicon" page on dlang.org since it apparently isn't obvious.

That's a good idea, but as you said, not here.